builtin/mainmenu/settings/components.lua

```diff
@@ -126,6 +126,40 @@ end, is_valid_number)
 make.string = make_field(tostring, nil)
 
 
+function make.slider(setting)
+	local steps = tonumber(setting.steps) or 100
+
+	return {
+		info_text = setting.comment,
+		setting = setting,
+
+		get_formspec = function(self, avail_w)
+			local value = core.settings:get(setting.name) or setting.default
+			local scrollbar_value = value/(setting.max - setting.min)*steps - setting.min
+
+			self.resettable = true
+
+			local setting_label = ("label[0,0.25;%s]"):format(
+				setting.readable_name)
+			local scrollbar_options = ("scrollbaroptions[min=0;max=%f;smallstep=1;largestep=10]"):format(
+				steps)
+			local scrollbar_formspec = ("scrollbar[0,0.5;%f,0.4;horizontal;%s;%f]"):format(
+				avail_w - 1, setting.name, scrollbar_value)
+			local formspec = setting_label..scrollbar_options..scrollbar_formspec
```

Comment:
Why not use [`table.concat`](https://www.lua.org/manual/5.1/manual.html#pdf-table.concat)? That would make sense if you want to add more (e.g. value display/input box) to this field.

Reply:
Rest of code in this file also does this so I guess it's better for consistency.